sgpt/utils.py

```diff
@@ -19,6 +20,18 @@ def get_mode(cls, shell, code) -> "CompletionModes":
         return CompletionModes.NORMAL
 
 
+def get_shell_name() -> str:
+    """
+    Get the name of the shell.
+
+    :return: Shell name.
+    """
+    proc = psutil.Process(os.getppid())
+    return next(
+        p.name() for p in proc.parents() if "sh" in p.name() or "cmd" in p.name()
```

Comment:
This will not work with `zsh` or `bash` or other non `sh` shells, it will also crash sgpt.

Reply:
I am not sure but it works in my zsh shell, the os is Ubuntu.
![image](https://user-images.githubusercontent.com/23721828/230791624-d76e979d-bbe5-4882-b6b8-d6324bcc87f1.png)

Reply:
Ubuntu:
```
proc = psutil.Process(pid=459, name='bash', started='19:24:23')
AttributeError: 'Process' object has no attribute 'parents'
```

MacOS:
```
proc = psutil.Process(pid=1473, name='zsh', status='running', started='01:54:35')
StopIteration
```

Reply:
Okay, I think this may results from that I did add the current proc itself into the list, and I fix it now.